src/org/sosy_lab/java_smt/solvers/princess/PrincessAbstractProver.java

```diff
@@ -74,19 +80,29 @@ protected PrincessAbstractProver(
    */
   @Override
   public boolean isUnsat() throws SolverException {
+    try {
+      Generator.dumpSMTLIB2();
+    } catch (IOException pE) {
+      throw new RuntimeException(pE);
+    }
     Preconditions.checkState(!closed);
     wasLastSatCheckSat = false;
-    final Value result = api.checkSat(true);
-    if (result.equals(SimpleAPI.ProverStatus$.MODULE$.Sat())) {
-      wasLastSatCheckSat = true;
-      return false;
-    } else if (result.equals(SimpleAPI.ProverStatus$.MODULE$.Unsat())) {
-      return true;
-    } else if (result.equals(SimpleAPI.ProverStatus$.MODULE$.OutOfMemory())) {
-      throw new SolverException(
-          "Princess ran out of stack or heap memory, try increasing their sizes.");
+    if (useBinary) {
+      binaryModel.getOutput();
```

Comment:
Here we use the Princess binary to read back the SMTLIB2 script we just dumped to `Out.smt2`. The output from the binary is then redirected to a file called `Model.smt2`, and later read back in to parse the result. We may look into using pipes instead to avoid problems when multiple solver instances are used simultaneously.